src/ledger/LedgerManagerImpl.cpp

```diff
@@ -739,17 +739,35 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     // step 4
     mApp.getBucketManager().forgetUnreferencedBuckets();
 
-    // Maybe sleep for parameterized amount of time in simulation mode
-    auto sleepFor = std::chrono::microseconds{
-        mApp.getConfig().OP_APPLY_SLEEP_TIME_FOR_TESTING * txSet->sizeOp()};
-    std::chrono::microseconds applicationTime =
-        closeLedgerTime.checkElapsedTime();
-    if (applicationTime < sleepFor)
+    if (!mApp.getConfig().OP_APPLY_SLEEP_TIME_FOR_TESTING.empty())
     {
-        sleepFor -= applicationTime;
-        CLOG_DEBUG(Perf, "Simulate application: sleep for {} microseconds",
-                   sleepFor.count());
-        std::this_thread::sleep_for(sleepFor);
+        // Sleep for a parameterized amount of time in simulation mode
+        int sample = rand_uniform(1, 100);
+        uint32_t chosenDuration = 0;
+        // Pick the i-th duration if only if the sample above lies within
+        // [percentage_0 + ... + percentage_{i-1} + 1, percentage_0 + ... +
+        // percentage_i] As the sum of the percentages equals 100, this gives us
+        // the i-th duration percentage_i percent of the time.
+        for (auto const& p : mApp.getConfig().OP_APPLY_SLEEP_TIME_FOR_TESTING)
+        {
+            sample -= p.first;
+            if (sample <= 0)
+            {
+                chosenDuration = p.second;
+                break;
+            }
+        }
+        auto sleepFor =
+            std::chrono::microseconds{chosenDuration * txSet->sizeOp()};
```

Comment:
do we want this or should it pick a random duration for each op separately?

Reply:
I updated it to pick a random duration for each op since I thought that it'll be closer to what the real network is likely doing.

But then after updating the code I remembered that it may not matter since loadgen currently includes only one Payment operation in a transaction. I'm not quite sure what would be the best approach here. Let me know what you think.